R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+#'   keyword is permitted only for queries which are defined by a `smq_select()`
+#'   or `sdg_select()` object.
+#'
+#' @param id The variable `<prefix>CD` is set to the specified value for
+#'   observations matching the query definition if the parameter is specified.
+#'   Otherwise the variable is not created. If the `auto` keyword is specified,
+#'   the variable is set to the id of the query in the SMQ/SDG database.
+#'
+#'   *Permitted Values*: A integer scalar or the `auto` keyword. The `auto`
+#'   keyword is permitted only for queries which are defined by a `smq_select()`
+#'   or `sdg_select()` object.
+#'
+#' @param add_scope_num The variable `<prefix>SCN` is added to the output
+#'   dataset if the parameter is specified. Otherwise the variable is not
+#'   created.
+#'
+#'   The value is set to `1` if the scope is broad. Otherwise, it is set to `2`.
+#'
+#'   If the parameter is set to `TRUE`, the definition must be a `smq_select()`
+#'   object.
+#'
+#'   *Default*: `FALSE`
+#'
+#'   *Permitted Values*: `TRUE`, `FALSE`
+#'
+#' @param definition Definition of terms belonging to the query
+#'
+#'   There are four different ways to define the terms:
+#'
+#'   * A `smq_select` object is specified to select a query from the SMQ
+#'     database. The database contains Standard MedDRA Queries (SMQs) and AE
+#'     Grouping Terms (AEGTs).
+#'
+#'   * A `sdg_select` object is specified to select a query from the SDG
+#'     database.
+#'
+#'   * A data frame with columns `TERM_LEVEL` and `TERM_NAME` or `TERM_ID` can
+#'     be specified to define the terms of the query. The `TERM_LEVEL` should be
+#'     set to the name of the variable which should be used to select the terms,
+#'     e.g., `"AEDECOD"` or `"AELLTCD"`. `TERM_LEVEL` does not need to be
+#'     constant within a query. For example a query can be based on `AEDECOD`
+#'     and `AELLT`.
+#'
+#'     If `TERM_LEVEL` refers to a character variable, `TERM_NAME` should be set
+#'     to the value the variable. If it refers to a numeric variable, `TERM_ID`
+#'     should be set to the value of the variable. If only character variables
+#'     or only numeric variables are used, `TERM_ID` or `TERM_NAME` respectively
+#'     can be omitted.
+#'
+#'   * A list of data frames and `smq_select` objects can be specified to define
+#'     a customized query based on custom terms and SMQs. The data frames must
+#'     have the same structure as described for the previous item.
+#'
+#'   *Permitted Values*: a `smq_select` object, a `sdg_select` object, a data
+#'   frame, or a list of data frames and `smq_select` objects.
+#'
+#' @details
+#'
+#' The definition includes
+#'
+#' * the variables which should be created for the query, e.g., `SMQ02NAM` and
+#' `SMQ02SC`,
+#'
+#' * the values which should be assigned to the variables, and
+#'
+#' * which terms belong to the query.
+#'
+#'
+#' @author Stefan Bundfuss
+#'
+#' @seealso [create_query_data()]
```

Comment:
shoud we add smq_select and sdg_select object here too?

Reply:
Yes, added